[Zurb] Work package Styling Quick wins

floriank opened the pull request:
This will fix some minor issues with the work package views.

These fixes will act as a band aid until the new work packages views are finished and will therefore not be in use for long.

Meets https://community.openproject.org/work_packages/19073

Commit:
add a line between description and the work package relations

Commit:
remove grey background for new work package form

Commit:
make attachments form openable by providing a local variable

By default, open the attachment for on the work package form

Commit:
make the attachments and watchers tables align properly

This applies only to the work package related views

Commit:
uses nowrap for labels to collapse them properly

Commit:
upgrade the labels in formbuilder

this will add a title attribute to labels to make them readable when collapsed

TeatroIO commented:
I've prepared a stage to preview changes. <a href='http://floriank-fix-19073-workpackage-quick-wins.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://floriank-fix-19073-workpackage-quick-wins.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

Commit:
please the hound

[ci skip]

myabc commented on spec/lib/tabular_form_builder_spec.rb:
```diff
@@ -504,15 +504,21 @@
+               title="Name">
+               Name
+        </label>
+      }.squish
```
@floriank you shouldn't need `.squish` here.

ulferts commented on app/assets/stylesheets/content/_forms.sass:
```diff
@@ -242,6 +242,7 @@ fieldset.form--fieldset
   @include grid-content(2)
   @include label-style
   padding:      0 1rem 0 0
+  white-space:  nowrap
```
@floriank this would be better placed inside the label-style mixin.

ulferts commented on app/views/work_packages/show.html.erb:
```diff
@@ -52,7 +52,7 @@ See doc/COPYRIGHT.rdoc for more details.
 
   <% if work_package.description? %>
-    <hr />
+    <hr>
```
The requirements of https://community.openproject.org/work_packages/19073 state to also have an `hr` below the heading.

floriank replied:
? This would belong to typography and i removed it from the condensed list of quick wins (see commentary in the ticket)

ulferts commented on lib/tabular_form_builder.rb:
```diff
@@ -55,6 +55,7 @@ def #{selector}(field, options = {}, *args)
 
   def label(method, text = nil, options = {}, &block)
     options[:class] = Array(options[:class]) + %w(form--label)
+    options[:title] = options[:title] || object.class.human_attribute_name(method)
```
@floriank as the styles are applied to all labels, all labels should hence benefit from having a title. For that we need to add this to the `styled_label_tag` helper method as well.

ulferts commented on spec/lib/tabular_form_builder_spec.rb:
```diff
@@ -504,15 +504,21 @@
       it 'should keep associated classes' do
         expect(output).to be_html_eql %{
-          <label class="sharknado form--label" for="user_name">Fear</label>
+          <label class="sharknado form--label" for="user_name" title="Fear">Fear</label>
```
The tests should also cover the default case where no title has been provided.

floriank replied:
Well it does - right above in the output test.

Or do you mean the case where I pass in a blank value for the title?

Commit:
remove superfluous call to squish

Commit:
move the whitespace attribute to label mixin

Commit:
update the form tag helper to allow for default title in label

floriank commented:
I am no entirely sure about what @houndci complains here

floriank commented:
![nqxm8pr](https://cloud.githubusercontent.com/assets/498241/6594902/5d38fef0-c7e5-11e4-8e78-85f4c458dd0c.jpg)

Commit:
fix all the things

Honestly, we are not using the case very often where a block is passed to the
styled_label_tag, so a change which expects a block passed will break almost
everything within the application.

Commit:
fix attachment test

the attachment fields are now shown directly, so the test no longer needs to
open the table.

Commit:
fix using the label method in conjunction with OpenStruct

Commit:
remove whitespace

[ci skip]
